sender_pool.go

```diff
@@ -219,10 +263,81 @@ func (p *LineSenderPool) IsClosed() bool {
 	return p.closed
 }
 
-// Len returns the numbers of cached LineSenders in the pool.
+// Len returns the number of LineSenders in the pool.
 func (p *LineSenderPool) Len() int {
 	p.mu.Lock()
 	defer p.mu.Unlock()
 
-	return len(p.senders)
+	return p.numSenders
+}
+
+func (ps *pooledSender) Table(name string) LineSender {
+	ps.wrapped.Table(name)
+	return ps
+}
+
+func (ps *pooledSender) Symbol(name, val string) LineSender {
+	ps.wrapped.Symbol(name, val)
+	return ps
+}
+
+func (ps *pooledSender) Int64Column(name string, val int64) LineSender {
+	ps.wrapped.Int64Column(name, val)
+	return ps
+}
+
+func (ps *pooledSender) Long256Column(name string, val *big.Int) LineSender {
+	ps.wrapped.Long256Column(name, val)
+	return ps
+}
+
+func (ps *pooledSender) TimestampColumn(name string, ts time.Time) LineSender {
+	ps.wrapped.TimestampColumn(name, ts)
+	return ps
+}
+
+func (ps *pooledSender) Float64Column(name string, val float64) LineSender {
+	ps.wrapped.Float64Column(name, val)
+	return ps
+}
+
+func (ps *pooledSender) StringColumn(name, val string) LineSender {
+	ps.wrapped.StringColumn(name, val)
+	return ps
+}
+
+func (ps *pooledSender) BoolColumn(name string, val bool) LineSender {
+	ps.wrapped.BoolColumn(name, val)
+	return ps
+}
+
+func (ps *pooledSender) AtNow(ctx context.Context) error {
+	err := ps.wrapped.AtNow(ctx)
+	if err != nil {
+		ps.dirty = true
+	}
+	return err
+}
+
+func (ps *pooledSender) At(ctx context.Context, ts time.Time) error {
+	err := ps.wrapped.At(ctx, ts)
+	if err != nil {
+		ps.dirty = true
+	}
+	return err
+}
+
+func (ps *pooledSender) Flush(ctx context.Context) error {
+	err := ps.wrapped.Flush(ctx)
+	if err != nil {
+		ps.dirty = true
+	}
+	return err
+}
+
+func (ps *pooledSender) Close(ctx context.Context) error {
+	if atomic.AddUint64(&ps.tick, 1)&1 == 1 {
+		return errors.New("double sender close")
```

Comment:
do we need to return this error? In the other senders, we allow multiple `Close` calls but just return `nil` if the sender is already closed:
https://github.com/questdb/go-questdb-client/blob/53a2d35445526db127746d8946357d2eedd1c0e0/tcp_sender.go#L139 and https://github.com/questdb/go-questdb-client/blob/53a2d35445526db127746d8946357d2eedd1c0e0/http_sender.go#L287

Reply:
With pooled senders double close is much more problematic as the sender may be in-use. So, I've changed all senders to error on double close in c6eb221fa7b31884fa763f8659717d30cc8867fe